api/src/test/java/com/epam/pipeline/acl/metadata/MetadataEntityApiServiceTest.java

```diff
@@ -0,0 +1,587 @@
+
+        assertThat(entityApiService.filterMetadata(metadataFilter)).isEqualTo(pagedResult);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyFilterMetadataWhenPermissionIsNotGranted() {
+        initAclEntity(folder);
+        doReturn(pagedResult).when(mockMetadataEntityManager).filterMetadata(metadataFilter);
+
+        assertThrows(AccessDeniedException.class, () -> entityApiService.filterMetadata(metadataFilter));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldGetMetadataKeysForAdmin() {
+        doReturn(metadataFields).when(mockMetadataEntityManager).getMetadataKeys(ID, TEST_STRING);
+
+        assertThat(entityApiService.getMetadataKeys(ID, TEST_STRING)).isEqualTo(metadataFields);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldGetMetadataKeysWhenPermissionIsGranted() {
+        initAclEntity(folder, AclPermission.READ);
+        doReturn(metadataFields).when(mockMetadataEntityManager).getMetadataKeys(ID, TEST_STRING);
+
+        assertThat(entityApiService.getMetadataKeys(ID, TEST_STRING)).isEqualTo(metadataFields);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyGetMetadataKeysWhenPermissionIsNotGranted() {
+        initAclEntity(folder);
+        doReturn(metadataFields).when(mockMetadataEntityManager).getMetadataKeys(ID, TEST_STRING);
+
+        assertThrows(AccessDeniedException.class, () -> entityApiService.getMetadataKeys(ID, TEST_STRING));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldGetMetadataFieldsForAdmin() {
+        doReturn(descriptions).when(mockMetadataEntityManager).getMetadataFields(ID);
+
+        assertThat(entityApiService.getMetadataFields(ID)).isEqualTo(descriptions);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldGetMetadataFieldsWhenPermissionIsGranted() {
+        initAclEntity(folder, AclPermission.READ);
+        doReturn(descriptions).when(mockMetadataEntityManager).getMetadataFields(ID);
+
+        assertThat(entityApiService.getMetadataFields(ID)).isEqualTo(descriptions);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyGetMetadataFieldsWhenPermissionIsNotGranted() {
+        initAclEntity(folder);
+        doReturn(descriptions).when(mockMetadataEntityManager).getMetadataFields(ID);
+
+        assertThrows(AccessDeniedException.class, () -> entityApiService.getMetadataFields(ID));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldUploadMetadataFromFileForAdmin() {
+        doReturn(metadataEntities).when(mockMetadataUploadManager).uploadFromFile(ID, file);
+
+        assertThat(entityApiService.uploadMetadataFromFile(ID, file)).isEqualTo(metadataEntities);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER, roles = ENTITIES_MANAGER_ROLE)
+    public void shouldUploadMetadataFromFileWhenPermissionIsGranted() {
+        initAclEntity(folder, AclPermission.READ);
```

Comment:
Seems like WRITE permission is required for the testing method to work. I'm not pretty sure how it passes right now but it should be investigated.